src/components/cloud-media/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { ImgHTMLAttributes } from 'react';
+import classNames from 'classnames';
+import { AdvancedImage, AdvancedVideo } from '@cloudinary/react';
+import { CloudinaryVideo } from '@cloudinary/url-gen';
+import { pad } from '@cloudinary/url-gen/actions/resize';
+import { auto } from '@cloudinary/url-gen/qualifiers/background';
+import { getCloudMedia, getHashFromIpfsUrl } from './utils';
+import { CloudinaryMedia, MediaType } from './types';
+
+export interface ComponentProperties
+  extends ImgHTMLAttributes<HTMLImageElement> {
+  className: string;
+  getCloudMedia: (
+    publicId: string,
+    prefix?: string
+  ) => Promise<CloudinaryMedia>;
+  getHashFromIpfsUrl: (url: string) => string;
+}
+
+interface State {
+  isLoaded: boolean;
+  cloudinaryMedia: CloudinaryMedia;
+  isLoading: boolean;
```

Comment:
is there any need for this piece of state? it *looks* to me like this just always corresponds to `state.cloudinaryMedia` being set? i generally like to avoid using explicit state for this kind of thing, when the thing that is loading ends up in state/as a property. i find that having two things that mean the same thing just gives an opportunity for things to get out of sync.